zerver/lib/streams.py

```diff
@@ -160,19 +185,13 @@ def create_stream_if_needed(
     )
 
     if can_remove_subscribers_group is None:
-        # TODO: Use default group instead of a fixed value.
-        can_remove_subscribers_group = NamedUserGroup.objects.get(
-            name=SystemGroups.ADMINISTRATORS, is_system_group=True, realm=realm
+        can_remove_subscribers_group = get_stream_permission_default_group(
+            "can_remove_subscribers_group", realm, creator=acting_user
         )
 
     if can_administer_channel_group is None:
-        can_administer_channel_group_default_name = Stream.stream_permission_group_settings[
-            "can_administer_channel_group"
-        ].default_group_name
-        can_administer_channel_group = NamedUserGroup.objects.get(
-            name=can_administer_channel_group_default_name,
-            realm=realm,
-            is_system_group=True,
+        can_administer_channel_group = get_stream_permission_default_group(
+            "can_administer_channel_group", realm, creator=acting_user
         )
```

Comment:
This one can probably also be turned into a loop, with a bit more care because one needs to look at `locals` like we do in view code. Probably lower priority than the others noted above; it's not necessarily bad to have a couple lines per setting here, in the core stream creation code.